models/issues/issue.go

```diff
@@ -480,6 +480,65 @@ func (issue *Issue) getLabels(ctx context.Context) (err error) {
 	return nil
 }
 
+func (issues IssueList) FilterValidByDoer(ctx context.Context, doer *user_model.User) (IssueList, error) {
+	repos, err := issues.LoadRepositories(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := repo_model.RepositoryList(repos).LoadOwners(ctx); err != nil {
+		return nil, err
+	}
+
+	issueList := issues[:0]
+	for _, issue := range issues {
+		if isIssueVisibleToDoer, err := issue.isIssueVisibleToDoer(ctx, doer); err != nil {
+			return nil, err
+		} else if isIssueVisibleToDoer {
+			issueList = append(issueList, issue)
+		}
+	}
+	return issueList, nil
+}
+
+// isIssueVisibleToDoer returns whether doer can view the issue
+func (issue *Issue) isIssueVisibleToDoer(ctx context.Context, doer *user_model.User) (bool, error) {
+	if perm, err := access_model.GetUserRepoPermission(ctx, issue.Repo, doer); err != nil {
+		return false, err
+	} else if !perm.HasAccess() {
+		return false, nil
+	}
+
+	if unit.TypeIssues.UnitGlobalDisabled() {
```

Comment:
This should not be checked here.

Reply:
I think we should check this, otherwise user can watch issues list when issue unit is globally disabled.
Maybe we can add it in `LoadIssuesFromBoard` or `LoadIssuesFromBoardList`?

Reply:
moved this check into `LoadIssuesFromBoard` in https://github.com/go-gitea/gitea/pull/22865/commits/8f3d51df1154bf13839027e3bfeda8d37fda43b1